File: features/lead/src/main/java/com/nivasafinance/features/leadlender/dto/UpdateLeadLenderRequest.java
package com.nivasafinance.features.leadlender.dto;

import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateLeadLenderRequest {
    
    private String lenderOfficeKey;
    
    @Valid
    private RmDetails rmDetails;
    
    @Valid
    private LoginDetails loginDetails;
    
    @Valid
    private ApprovedDetails approvedDetails;
    
    private String stage;
    
    private String remarks;
}

File: features/lead/src/main/java/com/nivasafinance/features/leadlender/dto/UpdateLeadLenderRequest.kt
package com.nivasafinance.features.leadlender.dto

import jakarta.validation.constraints.Pattern

data class UpdateLeadLenderRequest(
    val lenderOfficeKey: String? = null,
    val loginId: String? = null,
    val rmName: String? = null,
    @field:Pattern(regexp = "^[0-9]{10}$", message = "Mobile number must be exactly 10 digits")
    val rmMobileNumber: String? = null
)
